File: go/test/protocol/default_invoke_command.go
package protocol

type DefaultInvokeCommand struct {
	CommandName  *string          `toml:"command-name"`
	ExecutorID   *string          `toml:"executor-id"`
	RequestValue *string          `toml:"request-value"`
	Timeout      TestCaseDuration `toml:"timeout"`
}

func (invokeCommand DefaultInvokeCommand) GetCommandName() *string {
	if invokeCommand.CommandName == nil {
		return nil
	}

	commandName := *invokeCommand.CommandName
	return &commandName
}

func (invokeCommand DefaultInvokeCommand) GetExecutorID() *string {
	if invokeCommand.ExecutorID == nil {
		return nil
	}

	executorID := *invokeCommand.ExecutorID
	return &executorID
}

func (invokeCommand DefaultInvokeCommand) GetRequestValue() *string {
	if invokeCommand.RequestValue == nil {
		return nil
	}

	requestValue := *invokeCommand.RequestValue
	return &requestValue
}

func (invokeCommand DefaultInvokeCommand) GetTimeout() TestCaseDuration {
	return invokeCommand.Timeout
}

File: rust/azure_iot_operations_protocol/tests/metl/default_invoke_command.rs
use serde::Deserialize;

use crate::metl::test_case_duration::TestCaseDuration;

#[derive(Deserialize, Debug)]
pub struct DefaultInvokeCommand {
    #[serde(rename = "command-name")]
    pub command_name: Option<String>,

    #[serde(rename = "executor-id")]
    pub executor_id: Option<String>,

    #[serde(rename = "request-value")]
    pub request_value: Option<String>,

    #[serde(rename = "timeout")]
    pub timeout: Option<TestCaseDuration>,
}
